packages/php-wasm/universal/src/lib/load-php-runtime.ts

```diff
@@ -0,0 +1,221 @@
+/**
+ * Loads the PHP runtime with the given arguments and data dependencies.
+ *
+ * This function handles the entire PHP initialization pipeline. In particular, it:
+ *
+ * * Instantiates the Emscripten PHP module
+ * * Wires it together with the data dependencies and loads them
+ * * Ensures is all happens in a correct order
+ * * Waits until the entire loading sequence is finished
+ *
+ * Basic usage:
+ *
+ * ```js
+ *  const phpLoaderModule = await getPHPLoaderModule("7.4");
+ *  const php = await loadPHPRuntime( phpLoaderModule );
+ *  console.log(php.run(`<?php echo "Hello, world!"; `));
+ *  // { stdout: ArrayBuffer containing the string "Hello, world!", stderr: [''], exitCode: 0 }
+ * ```
+ *
+ * **The PHP loader module:**
+ *
+ * In the basic usage example, `phpLoaderModule` is **not** a vanilla Emscripten module. Instead,
+ * it's an ESM module that wraps the regular Emscripten output and adds some
+ * extra functionality. It's generated by the Dockerfile shipped with this repo.
+ * Here's the API it provides:
+ *
+ * ```js
+ * // php.wasm size in bytes:
+ * export const dependenciesTotalSize = 5644199;
+ *
+ * // php.wasm filename:
+ * export const dependencyFilename = 'php.wasm';
+ *
+ * // Run Emscripten's generated module:
+ * export default function(jsEnv, emscriptenModuleArgs) {}
+ * ```
+ *
+ * **PHP Filesystem:**
+ *
+ * Once initialized, the PHP has its own filesystem separate from the project
+ * files. It's provided by [Emscripten and uses its FS library](https://emscripten.org/docs/api_reference/Filesystem-API.html).
+ *
+ * The API exposed to you via the PHP class is succinct and universals
+ * await certain unintuitive parts of low-level FS interactions.
```

Comment:
I'm confused on this sentence, specifically "universals await" - I think there's a typo somewhere but I'm not sure what it is.

Reply:
Oops – mass find & replace doesn't always work as I intend :-) It was supposed to say `abstracts` and then there should be no `await` in the next line.

Reply:
Fixed in [5594e8e](https://github.com/WordPress/wordpress-playground/pull/214/commits/5594e8ead862e751285f9e41ccf553f0467896ee)